ledger-core/virtual/execute/delegatedtokenrequest.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package execute
+
+import (
+	"context"
+	"errors"
+
+	"github.com/insolar/assured-ledger/ledger-core/conveyor"
+	"github.com/insolar/assured-ledger/ledger-core/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/insolar/node"
+	"github.com/insolar/assured-ledger/ledger-core/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/network/messagesender"
+	messageSenderAdapter "github.com/insolar/assured-ledger/ledger-core/network/messagesender/adapter"
+	"github.com/insolar/assured-ledger/ledger-core/reference"
+	"github.com/insolar/assured-ledger/ledger-core/testutils/gen"
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/injector"
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+)
+
+type SMDelegatedTokenRequest struct {
+	// input arguments
+	Meta           *payload.Meta
+	RequestPayload payload.VDelegatedCallRequest
+
+	response *payload.VDelegatedCallResponse
+
+	// dependencies
+	pulseSlot     *conveyor.PulseSlot
+	messageSender messageSenderAdapter.MessageSender
+}
+
+/* -------- Declaration ------------- */
+
+var dSMDelegatedTokenRequestInstance smachine.StateMachineDeclaration = &dSMDelegatedTokenRequest{}
+
+type dSMDelegatedTokenRequest struct {
+	smachine.StateMachineDeclTemplate
+}
+
+func (*dSMDelegatedTokenRequest) InjectDependencies(sm smachine.StateMachine, _ smachine.SlotLink, injector *injector.DependencyInjector) {
+	s := sm.(*SMDelegatedTokenRequest)
+
+	injector.MustInject(&s.pulseSlot)
+	injector.MustInject(&s.messageSender)
+}
+
+func (*dSMDelegatedTokenRequest) GetInitStateFor(_ smachine.StateMachine) smachine.InitFunc {
+	return nil
+}
+
+/* -------- Instance ------------- */
+
+func (s *SMDelegatedTokenRequest) GetSubroutineInitState(_ smachine.SubroutineStartContext) smachine.InitFunc {
+	return s.Init
+}
+
+func (s *SMDelegatedTokenRequest) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
+	return dSMDelegatedTokenRequestInstance
+}
+
+func (s *SMDelegatedTokenRequest) Init(ctx smachine.InitializationContext) smachine.StateUpdate {
+	ctx.SetDefaultMigration(s.migration)
+	return ctx.Jump(s.stepRegisterBargeIn)
+}
+
+func (s *SMDelegatedTokenRequest) stepRegisterBargeIn(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	bargeInCallback := ctx.NewBargeInWithParam(func(param interface{}) smachine.BargeInCallbackFunc {
+		res, ok := param.(*payload.VDelegatedCallResponse)
+		if !ok || res == nil {
+			panic(throw.IllegalValue())
+		}
+		s.response = res
+
+		return func(ctx smachine.BargeInContext) smachine.StateUpdate {
+			return ctx.WakeUp()
+		}
+	})
+
+	callOutgoing := gen.UniqueIDWithPulse(s.pulseSlot.PulseData().PulseNumber)
```

Comment:
@bigbes @ruz . I'm not sure, but this looks wrong, you already have outgoing call in VCallRequest.

Reply:
I think that this must be unique id for BargeIn and it can't be == outgoing call in VCallRequest

Reply:
Please, create alias type:
```
type DelegationTokenAwaitKey struct {
    Outgoing reference.Global
}
```

and use it for delegatonToken key. You can add pulse if you'll need separate processing for diefferent pulses

Reply:
done